src/nvim/path.c

```diff
@@ -360,7 +360,7 @@ void add_pathsep(char *p)
   FUNC_ATTR_NONNULL_ALL
 {
   if (*p != NUL && !after_pathsep(p, p + strlen(p)))
-    strcat(p, PATHSEPSTR);
+    STRLCAT(p, PATHSEPSTR, MAXPATHL);
```

Comment:
Did you check all call sites of function `add_pathsep()` that the buffer is always big enough or not bigger than `MAXPATHL`?